### PR TITLE
BUG: Fix broken copy-paste

### LINKED ISSUE #243: Copy and pasting in plates stopped working

### PR DESCRIPTION
**Pull request by ElDeveloper**
The assignment of the `this.blankNames` property was being done
asynchronously on a per-cell basis. This worked fine if the user inputed
a value by hand, as it gave enough time for the request to return.
However, when someone copy-pasted text into the grid, the request
wouldn't return in time and the `this.blankNames` property wouldn't be
correctly assigned leading to a ReferenceError.

This is now fixed by getting the contents of `this.blankNames` before
initializing the whole grid. This way every cell has access to the data
before instantiation.

Fixes #243

### CODE REVIEW

File: labman/gui/static/js/plateViewer.js [hunk 1]
```diff
@@ -129,6 +129,13 @@ PlateViewer.prototype.initialize = function (rows, cols) {
                        cssClass: 'slick-header-column',
                        headerCssClass: 'full-height-header'}];
 
+  // Fetch the blank names before initializing the grid. This way we only make
+  // one request to get this data instead of one per cell instantiation.
+  var blanksRequest = $.ajax({
+    dataType: 'json',
+    url: '/sample/control?term=',
```
Comment (wasade) [inline labman/gui/static/js/plateViewer.js]: To make sure I understand this, by not requesting a term or set of terms, we get back all of the possible controls?

File: labman/gui/static/js/plateViewer.js [hunk 2]
```diff
@@ -536,13 +544,9 @@ function SampleCellEditor(args) {
...
 
-  this.blanknames = [];
-  $.ajax({
-    url: '/sample/control?term=',
-    success: function (result) {
-      that.blankNames = result;
-    }
-  });
+  // Get the parsed values from the request. Alternatively, if the request
+  // didn't complete yet use an empty list as a fallback.
```
Comment (wasade) [inline labman/gui/static/js/plateViewer.js]: Is it a failure if the request does not complete?

File: labman/gui/static/js/plateViewer.js (hunk 1 above)
Comment (wasade) [reply labman/gui/static/js/plateViewer.js]: Okay, thanks. Would it be possible to add a comment indicating as such as the behavior is implicit?

File: labman/gui/static/js/plateViewer.js (hunk 2 above)
Comment (wasade) [reply labman/gui/static/js/plateViewer.js]: I think so, yes. Is it possible to block until the ajax completes followed by a test of a status code?

Comment (ElDeveloper) [reply labman/gui/static/js/plateViewer.js]: Blocking requests are frowned upon and deprecated (or on their way out). This [link](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/Synchronous_and_Asynchronous_Requests) has information (see under Synchronous Requests).

For now, I'll add some error handling and let the user know if they need to reload the page to try again.

File: labman/gui/static/js/plateViewer.js (hunk 1 above)
Comment (ElDeveloper) [reply labman/gui/static/js/plateViewer.js]: Will do.

File: labman/gui/static/js/plateViewer.js (hunk 2 above)
Comment (ElDeveloper) [reply labman/gui/static/js/plateViewer.js]: I've added some error handling (a message at the top of the screen describing the problem and a possible workaround). Note that for this test I modified the URI so the request pointed to a non-existant route hence the 404.

screen shot 2018-07-25 at 12 50 27 pm

Comment (wasade) [reply labman/gui/static/js/plateViewer.js]: :+1:

**Comment by wasade** [comment]
I'm :+1: pending travis. @AmandaBirmingham, would you like a second review or good if green w/ travis?

**Comment by AmandaBirmingham** [comment]
@wasade I looked over the changes and the discussion, and I don't have any useful comments.  If travis green-lights this, so do I :)

**Comment by ElDeveloper** [comment]
Thanks!

**Pull request merged**